windows/src/support/texteditor/FontSelect.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * TextEditor is copyright (C) SIL International. MIT License.
+ * Implementation of the FontSelect Class
+ */
+
+#include "framework.h"
+#include "FontSelect.h"
+#include "commdlg.h"
+
+font_select::font_select()
+{
+  default_font();
+}
+
+font_select::font_select(HFONT current_font)
+{
+
+	if (m_hfont != NULL) {
+		::DeleteObject(m_hfont);
+  }
+
+	// store the font current font for the calling application
+	m_hfont = current_font;
+  // We still set the default value but can now drop back to the current font
+	default_font();
+}
+
+font_select::~font_select()
+{
+	if (m_hfont != NULL) {
+		::DeleteObject(m_hfont);
+  }
+}
+
+void font_select::default_font()
+{
+  // Set the logical font information as "Charsis SIL" size 14
+	// A future enhancement could load default from a configuration file
+	LOGFONT log_font;
+	log_font.lfHeight = -19; // size 14
+	log_font.lfWidth = 0;
+	log_font.lfEscapement = 0;
+	log_font.lfOrientation = 0;
+	log_font.lfWeight = FW_NORMAL;
+	log_font.lfItalic = 0;
+	log_font.lfUnderline = 0;
+	log_font.lfStrikeOut = 0;
+	log_font.lfCharSet = 0;
+	log_font.lfOutPrecision = OUT_STRING_PRECIS;
+	log_font.lfClipPrecision = CLIP_STROKE_PRECIS;
+	log_font.lfQuality = DEFAULT_QUALITY;
+	log_font.lfPitchAndFamily = FF_SWISS | VARIABLE_PITCH;
+	wcscpy_s(log_font.lfFaceName, L"Charsis SIL");
```

Comment:
The font is Charis SIL, not Charsis SIL :grin:

```suggestion
	wcscpy_s(log_font.lfFaceName, L"Charis SIL");
```